pinot-common/src/main/java/org/apache/pinot/common/utils/config/QueryWorkloadConfigUtils.java

```diff
@@ -211,24 +215,167 @@ public static List<String> validateQueryWorkloadConfig(QueryWorkloadConfig confi
         if (enforcementProfile == null) {
           errors.add(prefix + "enforcementProfile cannot be null");
         } else {
-           if (enforcementProfile.getCpuCostNs() < 0) {
+          long enforcementCpu = enforcementProfile.getCpuCostNs();
+          long enforcementMem = enforcementProfile.getMemoryCostBytes();
+           if (enforcementCpu < 0) {
              errors.add(prefix + ".enforcementProfile.cpuCostNs cannot be negative");
            }
-           if (enforcementProfile.getMemoryCostBytes() < 0) {
-               errors.add(prefix + ".enforcementProfile.memoryCostBytes cannot be negative");
+           if (enforcementMem < 0) {
+             errors.add(prefix + ".enforcementProfile.memoryCostBytes cannot be negative");
            }
-        }
-        // Validate PropagationScheme
-        PropagationScheme propagationScheme = nodeConfig.getPropagationScheme();
-        if (propagationScheme == null) {
-          errors.add(prefix + ".propagationScheme cannot be null");
-        } else {
-          if (propagationScheme.getPropagationType() == null) {
-            errors.add(prefix + ".propagationScheme.type cannot be null");
+          // Validate PropagationScheme
+          PropagationScheme propagationScheme = nodeConfig.getPropagationScheme();
+          if (propagationScheme == null) {
+            errors.add(prefix + ".propagationScheme cannot be null");
+          } else {
+            PropagationScheme.Type propagationType = propagationScheme.getPropagationType();
+            if (propagationType == null) {
+              errors.add(prefix + ".propagationScheme.type cannot be null");
```

Comment:
Why are we not continuing here to the next nodeConfig when we hit errors? Is the idea to collect all errrors and surface them in one-go? 

The reason I ask is - If propagationType is null how do you even validate the propagationEntities meaningfully?